core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -109,6 +113,95 @@ public DefaultRequestLog(RequestContext ctx) {
         this.ctx = requireNonNull(ctx, "ctx");
     }
 
+    @Override
+    public void addChild(RequestLog child) {
+        checkState(!hasLastChild, "last child is already added");
+        requireNonNull(child, "child");
+        if (children == null) {
+            // first child's all request-side logging events are propagated immediately to the parent
+            children = new ArrayList<>();
+            propagateRequestSideLog(child);
+        }
+        children.add(child);
+    }
+
+    private void propagateRequestSideLog(RequestLog child) {
+        child.addListener(log -> startRequest(log.channel(), log.sessionProtocol(), log.host()), REQUEST_START);
```

Comment:
Could we set the start time explicitly so that the start time of the parent request log is same with that of the first child? You could add a private variant of `startRequest()` which accepts the start time as a parameter.